src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -3141,6 +3141,30 @@ public static void DCS_KnownTypeMethodName()
         Assert.Equal(value.emps[1].Name, actual.emps[1].Name);
     }
 
+    [Fact]
+    public static void DCS_NativeDll()
+    {
+        NativeDllWrapper.CallIntoNativeDll();
+    }
+
+    [Fact]
+    public static void DCS_ReadOnlyDictionaryCausingDuplicateInvalidDataContract()
+    {
+        var dict = new Dictionary<string, int>();
+        dict["Foo"] = 1;
+        dict["Bar"] = 2;
+        ReadOnlyDictionary<string, int> value = new ReadOnlyDictionary<string, int>(dict);
+        var deserializedValue = SerializeAndDeserialize(value, @"<ReadOnlyDictionaryOfstringint xmlns=""http://schemas.datacontract.org/2004/07/System.Collections.ObjectModel"" xmlns:i=""http://www.w3.org/2001/XMLSchema-instance""><_dictionary xmlns:a=""http://schemas.microsoft.com/2003/10/Serialization/Arrays""><a:KeyValueOfstringint><a:Key>Foo</a:Key><a:Value>1</a:Value></a:KeyValueOfstringint><a:KeyValueOfstringint><a:Key>Bar</a:Key><a:Value>2</a:Value></a:KeyValueOfstringint></_dictionary></ReadOnlyDictionaryOfstringint>");
+
+        Assert.StrictEqual(value.Count, deserializedValue.Count);
+        Assert.StrictEqual(value["Foo"], deserializedValue["Foo"]);
+        Assert.StrictEqual(value["Bar"], deserializedValue["Bar"]);
+
+        DataContractJsonSerializer dcjs = new DataContractJsonSerializer(typeof(ReadOnlyDictionary<string, int>));
```

Comment:
The test code is incomplete. The original test probably was also incomplete.

Can you please complete this test and move this part to the DCJS test project?